internal/providersdkv2/resource_vault_cluster.go

```diff
@@ -1239,6 +1379,35 @@ func getObservabilityConfig(propertyName string, d *schema.ResourceData) (*vault
 	return getValidObservabilityConfig(config)
 }
 
+// if http observability information is provided, this function ensures that authentication fields are valid and returns the authentication method used
+func validateHTTPAuth(httpBasicUser, httpBasicPassword, httpBearerToken string) (*vaultmodels.HashicorpCloudVault20201125HTTPBearerAuth, *vaultmodels.HashicorpCloudVault20201125HTTPBasicAuth, diag.Diagnostics) {
+	// only one of basic or bearer authentication should be submitted
+	if httpBearerToken != "" && (httpBasicUser != "" || httpBasicPassword != "") {
+		return nil, nil, diag.Errorf("http configuration is invalid: either the basic or bearer authentication method can be submitted, but not both")
+	} else if httpBasicUser != "" && httpBasicPassword == "" || httpBasicUser == "" && httpBasicPassword != "" {
+		// http basic requires both the username and password to be filled
+		return nil, nil, diag.Errorf("http configuration is invalid: basic authentication requires username and password")
+	}
+
+	if httpBearerToken != "" {
+		httpBearerAuth := &vaultmodels.HashicorpCloudVault20201125HTTPBearerAuth{
+			Token: httpBearerToken,
+		}
+		return httpBearerAuth, nil, nil
+	}
+
+	if httpBasicUser != "" && httpBasicPassword != "" {
```

Comment:
nit: I think the `||` operator would make the intent clearer here, or even just checking for User as we'd be relying on  either both being set or both being empty